app/scripts/controllers/detect-tokens.js

```diff
@@ -183,14 +188,54 @@ export default class DetectTokensController {
    * in case of address change or user session initialization.
    *
    */
+  /* eslint-disable no-undef */
   restartTokenDetection() {
     if (!(this.isActive && this.selectedAddress)) {
       return;
     }
     this.detectNewTokens();
+    if (isManifestV3) {
+      chrome.alarms.clear(DETECT_TOKEN_ALARM);
+      chrome.alarms.onAlarm.removeListener(this.alarmListener);
+    }
     this.interval = DEFAULT_INTERVAL;
   }
 
+  /**
+   * Restart/Reschedule token detection polling period and call detectNewTokens
+   * in case of address change or user session initialization.
+   *
+   */
+  /* eslint-disable no-undef */
+  rescheduleTokenDetectionPollingInMV3() {
+    if (!(this.isActive && this.selectedAddress)) {
+      return;
+    }
+    chrome.alarms.getAll((alarms) => {
+      const hasAlarm = alarms.some(
+        (alarm) => alarm.name === DETECT_TOKEN_ALARM,
+      );
+      if (!hasAlarm) {
+        chrome.alarms.create(DETECT_TOKEN_ALARM, {
+          delayInMinutes: DEFAULT_INTERVAL_MV3,
+          periodInMinutes: DEFAULT_INTERVAL_MV3,
+        });
+      }
+    });
+    chrome.alarms.onAlarm.addListener(this.alarmListener);
+  }
+
+  alarmListener = () => {
+    chrome.alarms.getAll((alarms) => {
+      const hasAlarm = alarms.some(
```

Comment:
Nit:  This `alarms.some` is used twice; I'm wondering if we can put this in a helper function.